### PR TITLE
Script fixes, Init UE Tau Fix

### PR DESCRIPTION
**Pull request by asreek**
1. Added support in the c++ class generation script to support counter types as data members.
2. Fixed a bug in state class template file to add unique headers for "next state" classes, in case the same next-state entry exists for multiple events.
3. Removed the duplicate action handler in TAU state machine to process the TAU. The default action handler is invoked in this case, which consumes the tau request message.

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: src/mme-app/actionHandlers/attachActionHandlers.cpp [hunk 1]
```diff
@@ -612,7 +612,7 @@ ActStatus ActionHandlers::send_esm_info_req_to_ue(SM::ControlBlock& cb)
 	esmreq.dl_seq_no = ue_ctxt->getDwnLnkSeqNo();
 	memcpy(&(esmreq.int_key), &((ue_ctxt->getUeSecInfo().secinfo_m).int_key),
 			NAS_INT_KEY_SIZE);
-	ue_ctxt->setDwnLnkSeqNo(esmreq.dl_seq_no+1);
+	ue_ctxt->incrementDwnLnkSeqNo();
```
Comment (thakurajayL) [inline src/mme-app/actionHandlers/attachActionHandlers.cpp:615]: @badhrinathpa This API may not be good right ? Given that sequence number is not just counter but more than that.

File: src/mme-app/actionHandlers/s1HandoverActionHandlers.cpp [hunk 2]
```diff
@@ -395,6 +395,60 @@ ActStatus ActionHandlers::send_s1_rel_cmd_to_src_enb_for_ho(ControlBlock& cb)
...
+    const struct tauReq_Q_msg &tauReq = (msgData_p->msg_data.tauReq_Q_msg_m);
+    ue_ctxt->incrementUpLnkSeqNo();
+
+    //TAI and CGI obtained from s1ap ies.
+    //Convert the plmn in s1ap format to nas format
+    //before storing in ue context/sending in tai list of tau response.
+    MmeCommonUtils::formatS1apPlmnId(const_cast<PLMN*>(&tauReq.tai.plmn_id));
+    MmeCommonUtils::formatS1apPlmnId(
+            const_cast<PLMN*>(&tauReq.eUtran_cgi.plmn_id));
+    s1HoPrCtxt->setTargetTai(Tai(tauReq.tai));
```
Comment (thakurajayL) [inline src/mme-app/actionHandlers/s1HandoverActionHandlers.cpp:446]: What are the different ULI parameter's we should handle ?

File: src/mme-app/actionHandlers/attachActionHandlers.cpp (hunk 1 above)
Comment (badhrinathpa) [reply src/mme-app/actionHandlers/attachActionHandlers.cpp:615]: I have added the right way to update sequence numbers and counters in the security branch. 
We can keep the metrics generator code part though. that will be useful.

**Comment by asreek** [comment]
Closing this, as new commits are not showing up in the PR.

File: src/mme-app/actionHandlers/s1HandoverActionHandlers.cpp (hunk 2 above)
Comment (asreek) [reply src/mme-app/actionHandlers/s1HandoverActionHandlers.cpp:446]: ULI in gtpv2? For 4g, TAI, eCGI and macro enbId are applicable.

File: src/mme-app/actionHandlers/attachActionHandlers.cpp (hunk 1 above)
Comment (asreek) [reply src/mme-app/actionHandlers/attachActionHandlers.cpp:615]: Ok. Retained the script changes. Reverted the Action handler changes.
Please review https://github.com/omec-project/Nucleus/pull/37

**Pull request closed without merging**